engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/savedGames/GameProvider.java

```diff
@@ -110,21 +111,42 @@ public static boolean isSavesFolderEmpty() {
     }
 
     /**
-     * Generates the game name based on the game number of the last saved game
+     * Generates the game name based on the game number of the last saved game Uses {@link
+     * GameProvider#DEFAULT_GAME_NAME_PREFIX} for resolve.
      */
     public static String getNextGameName() {
+        String nextGameName = getNextGameName(DEFAULT_GAME_NAME_PREFIX);
+        if (nextGameName.equals(DEFAULT_GAME_NAME_PREFIX)) {
+            return nextGameName + 1;
+        }
+        return nextGameName;
+    }
+
+    /**
+     * Generates the game name based on the game number of the last saved game
+     *
+     * @param gameName will to use as game prefix, if saves contains this game name
+     * @return next game name with number. Example: "Game" -> "Game 1", "Game 1 -> Game 2"
+     */
+    public static String getNextGameName(String gameName) {
         int gameNumber = 1;
+        String name = gameName;
         for (GameInfo info : GameProvider.getSavedGames()) {
-            if (info.getManifest().getTitle().startsWith(DEFAULT_GAME_NAME_PREFIX)) {
-                String remainder = info.getManifest().getTitle().substring(DEFAULT_GAME_NAME_PREFIX.length());
+            if (info.getManifest().getTitle().startsWith(name)) {
```

Comment:
This prefix match and handling looks a bit shaky to me. We probably should define way more tests to define what we expect here...